### PR TITLE
feat(DEV-1252): EKS ArgoCD Environments

### PR DESCRIPTION
**Pull request by milldr**
## what
- Consolidating all required workflows into a single repo

## why
- Easier maintenance and simplicity 

## references
- DEV-1252
- https://github.com/cloudposse/github-actions-workflows-docker-ecr-eks-helm-argocd
- https://github.com/cloudposse/github-actions-workflows

### CODE REVIEW

File: .github/workflows/workflow-cd-preview-argocd.yml [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+        uses: actions/checkout@v3
+
+      - name: Environment Info
+        uses: cloudposse/github-action-interface-environment@0.3.0
+        id: environment
+        with:
+          implementation_repository: 'cloudposse/infra-live'
+          implementation_path: '.github/environments/'
+          implementation_file: 'eks-argocd.yml'
+          implementation_ref: 'DEV-1252/self-contained-workflows' # TODO revert to main
```
Comment (milldr) [inline .github/workflows/workflow-cd-preview-argocd.yml:164]: todo

File: .github/workflows/workflow-cd-preview-argocd.yml [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+      - name: Environment Info
+        uses: cloudposse/github-action-interface-environment@0.3.0
+        if: ${{ steps.deployment.outputs.id != '' }}
+        id: environment
+        with:
+          implementation_repository: 'cloudposse/infra-live'
+          implementation_path: '.github/environments/'
+          implementation_file: 'eks-argocd.yml'
+          implementation_ref: 'DEV-1252/self-contained-workflows' # TODO revert to main
```
Comment (milldr) [inline .github/workflows/workflow-cd-preview-argocd.yml:231]: todo

**Comment by milldr** [comment]
todo:
```console
Failed sync attempt to 2cefc09f33a332f66c6d179942c763e22c6bfe4d: one or more objects failed to apply, reason: Service "example-app-monochart" is invalid: metadata.labels: Invalid value: "dev-1252-self-contained-workflows.example-app-on-eks-with-argocd": must be no more than 63 characters,Deployment.apps "example-app-monochart" is invalid: metadata.labels: Invalid value: "dev-1252-self-contained-workflows.example-app-on-eks-with-argocd": must be no more than 63 characters,Ingress.extensions "example-app-monochart-default" is invalid: metadata.labels: Invalid value: "dev-1252-self-contained-workflows.example-app-on-eks-with-argocd": must be no more than 63 characters (retried 5 times).
```

**Comment by milldr** [comment]
deployed successfully:
CleanShot 2023-12-07 at 15 07 22

**Review by Benbentwo — APPROVED** [review]

**Pull request merged**